### ISSUE TITLE
Rule Change: Allow restricting only default imports in no-restricted-imports

### ISSUE THREAD

**Issue by joshkel** [eslint/eslint#15895]
### What rule do you want to change?

no-restricted-imports

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
import * as React from 'react';
```

### What does the rule currently do for this code?

When used with a no-restricted-imports configuration of

```
[
      'error',
      { paths: [{ name: 'react', importNames: ['default'] }]}
]
```

ESLint rejects with the following error:

```
* import is invalid because 'default' from 'react' is restricted. 
```

### What will the rule do after it's changed?

Add an option to disallow the default import only, while still allowing wildcard/namespace imports. Maybe something like this?

```
[
      'error',
      { paths: [{ name: 'react', importNames: ['default'], allowEverything: true }]}
]
```

(I'm not sure about the option name: I noticed that no-restricted-imports.js uses "everything" for its message name, but the node name and React docs calls it "namespace," and "namespace" or "wildcard" seems clearer to the end user.)

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

The React team has started discouraging the use of the default import, but they say that the wildcard/namespace import is still valid. (See [here](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports) and [here](https://twitter.com/dan_abramov/status/1308739731551858689).) I looked into using no-restricted-imports to enforce this, but it doesn't appear to allow that: if _anything_ is restricted, then no-restricted-imports always disallows '*'. There may be cases where individual imports (or, as in React's case, the default import) should be restricted to enforce a consistent coding style, even if the restricted import could technically be accessed via the imported namespace.

**Comment by nzakas** [comment]
This is interesting. I’m not sure if a namespace import should or should not count towards blocking exported names. This could be a good candidate for a new option (I’d suggest maybe excludeNamespaceImports for clarity).

Id like some other opinions from the team.

**Comment by oliviertassinari** [comment]
It might be stale but the problem is still real.

**Comment by oliviertassinari** [comment]
A good alternative that works: https://github.com/mui/material-ui/pull/34878#issuecomment-1290679529
